earth_enterprise/src/portableserver/build.py

```diff
@@ -0,0 +1,252 @@
+        self.resources_dir = os.path.join(
+            self.source_dir, 'portableserver', 'resources')
+        self.server_dir = os.path.join(self.package_dir, 'server')
+        self.tar_package_name = \
+            'portableserver-{}-{}'.format(self.platform, self.version_string)
+        self.zip_package_name = self.tar_package_name
+        if self.platform == 'windows':
+            self.should_create_tar_package = False
+            self.should_create_zip_package = True
+        else:
+            self.should_create_tar_package = True
+            self.should_create_zip_package = False
+
+    def build(self):
+        """Builds and packages Portable server."""
+
+        shutil.rmtree(self.build_dir, ignore_errors=True)
+        ensure_directory(self.build_dir)
+        ensure_directory(self.package_dir)
+        ensure_directory(self.server_dir)
+        self.obtain_server_resources()
+        self.build_fileunpacker()
+        self.obtain_sample_globes()
+        if self.should_create_tar_package:
+            self.create_tar_package()
+        if self.should_create_zip_package:
+            self.create_zip_package()
+
+    def clean(self):
+        """Deletes build files and directories."""
+
+        shutil.rmtree(self.build_dir, ignore_errors=True)
+
+    def obtain_server_resources(self):
+        """Builds <server/>."""
+
+        # Copy from <fusion/portableglobe/servers/>:
+        servers_dir = os.path.join(
+            self.source_dir, 'fusion', 'portableglobe', 'servers')
+
+        exclude_entries = ['linux', 'mac', 'windows', 'fileunpacker']
+        entries = [f
+                   for f in os.listdir(servers_dir)
+                   if f.lower() not in exclude_entries]
+        copy_from_dir_to_dir(servers_dir, self.server_dir, entries=entries)
+        copy_from_dir_to_dir(
+            os.path.join(servers_dir, self.platform), self.server_dir)
+
+        # Copy from <maps/>:
+        copy_from_dir_to_dir(
+            os.path.join(self.source_dir, 'maps'),
+            os.path.join(self.server_dir, 'local', 'maps'),
+            exclude_entries=['SConscript'])
+
+        self.create_version_txt()
+
+        # Copy configuration files:
+        copy_from_dir_to_dir(self.resources_dir, self.server_dir)
+
+    def create_version_txt(self):
+        """Creates <server/local/version.txt>."""
+
+        self.get_version()
+        path = os.path.join(self.server_dir, 'local', 'version.txt')
+        with open(path, 'w') as output_file:
+            output_file.write(self.version_string)
+
+    def get_version(self):
+        """Parses version information, and sets member variables."""
+
+        if self.base_version is None:
+            path = os.path.join(self.source_dir, 'fusion_version.txt')
+            with open(path, 'r') as input_file:
+                while True:
+                    base_version = input_file.readline().strip()
+                    if not base_version.startswith('#'):
```

Comment:
Add check for empty string:

if base_version and not base_version.startswith('#'):